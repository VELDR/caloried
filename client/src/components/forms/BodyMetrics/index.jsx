import PropTypes from 'prop-types';
import { useState } from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';
import { connect, useDispatch } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import toast from 'react-hot-toast';
import { Cake, CalendarToday, Female, Male, SquareFoot, Wc } from '@mui/icons-material';

import { nextStep, setMetrics } from '@pages/SignUp/actions';
import { selectMetrics } from '@pages/SignUp/selectors';

import FormFooter from '@components/forms/FormFooter';
import FormHeader from '@components/forms/FormHeader';

import classes from './style.module.scss';

const BodyMetrics = ({ metrics, intl: { formatMessage } }) => {
  const dispatch = useDispatch();
  const [metricsData, setMetricsData] = useState({
    sex: metrics.sex || '',
    dob: metrics.dob || '',
    height: metrics.height || '',
    weight: metrics.weight || '',
  });

  const handleSexSelect = (sex) => {
    setMetricsData({ ...metricsData, sex });
  };

  const handleInputChange = (e) => {
    setMetricsData({ ...metricsData, [e.target.name]: e.target.value });
  };

  const validateMetrics = () => {
    const { sex, dob, height, weight } = metricsData;
    const today = new Date();
    const dobDate = new Date(dob);
    if (!sex) {
      toast.error(formatMessage({ id: 'app_error_sex' }));
      return false;
    }
    if (!dob || dobDate >= today) {
      toast.error(formatMessage({ id: 'app_error_weight' }));
      return false;
    }
    if (!height || height < 120 || height > 250) {
      toast.error(formatMessage({ id: 'app_error_weight' }));
      return false;
    }
    if (!weight || weight < 20 || weight > 400) {
      toast.error(formatMessage({ id: 'app_error_weight' }));
      return false;
    }
    return true;
  };

  const handleFormSubmit = () => {
    if (validateMetrics()) {
      dispatch(setMetrics(metricsData));
      dispatch(nextStep());
    }
  };
  return (
    <div className={classes.form}>
      <FormHeader
        title={formatMessage({ id: 'app_metrics_title' })}
        description={formatMessage({ id: 'app_metrics_description' })}
      />

      <div className={classes.metrics}>
        <div className={classes.attributes}>
          <div className={classes.sex}>
            <div className={classes.sex__title}>
              <Wc />
              <FormattedMessage id="app_biological_sex" />
            </div>
            <div className={classes.sex__option}>
              <div
                className={`${classes.sex__item} ${classes.sex__male} ${
                  metricsData.sex === 'male' ? classes.activeMale : ''
                }`}
                onClick={() => handleSexSelect('male')}
              >
                <Male />
                <div className={classes.sex__label}>
                  <FormattedMessage id="app_male" />
                </div>
              </div>
              <div
                className={`${classes.sex__item} ${classes.sex__female} ${
                  metricsData.sex === 'female' ? classes.activeFemale : ''
                }`}
                onClick={() => handleSexSelect('female')}
              >
                <Female />
                <div className={classes.sex__label}>
                  <FormattedMessage id="app_female" />
                </div>
              </div>
            </div>
          </div>
          <div className={classes.dob}>
            <div className={classes.dob__label}>
              <Cake /> <FormattedMessage id="app_dob" />
            </div>
            <div className={classes.wrapper}>
              <input
                type="date"
                name="dob"
                className={classes.dob__input}
                value={metricsData.dob}
                onChange={handleInputChange}
              />
              <CalendarToday className={classes.calendar} />
            </div>
          </div>
        </div>
        <div className={classes.measurements}>
          <div className={classes.measurements__title}>
            <SquareFoot /> <FormattedMessage id="app_measurements" />
          </div>
          <div className={classes.measurement}>
            <div className={classes.measurement__item}>
              <div className={classes.label}>
                <FormattedMessage id="app_height" />
              </div>
              <div className={classes.wrapper}>
                <input
                  type="number"
                  name="height"
                  className={classes.input}
                  value={metricsData.height}
                  onChange={handleInputChange}
                />
                <div className={classes.adornment}>cm</div>
              </div>
            </div>
            <div className={classes.measurement__item}>
              <div className={classes.label}>
                <FormattedMessage id="app_weight" />
              </div>
              <div className={classes.wrapper}>
                <input
                  type="number"
                  name="weight"
                  className={classes.input}
                  value={metricsData.weight}
                  onChange={handleInputChange}
                />
                <div className={classes.adornment}>kg</div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <FormFooter onContinue={handleFormSubmit} />
    </div>
  );
};

BodyMetrics.propTypes = {
  metrics: PropTypes.object,
  intl: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  metrics: selectMetrics,
});

export default injectIntl(connect(mapStateToProps)(BodyMetrics));
