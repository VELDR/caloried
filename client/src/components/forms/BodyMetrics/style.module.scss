@import '@styles/base';

.form {
  @include flex-column(0.75rem);
  height: 100%;
  @include apply-at-md-size {
    justify-content: space-between;
  }
  .metrics {
    @include flex-center-column;

    .attributes {
      @include flex-center-column;
      width: 100%;
      @include apply-at-md-size {
        @include flex-items-center;
        flex-direction: row;
        border-top: 1px solid var(--color-text-secondary);
        border-bottom: 1px solid var(--color-text-secondary);
      }
      .sex {
        @include flex-center-column(0.5rem);
        width: 100%;
        padding: 0.75rem;
        border-top: 1px solid var(--color-text-secondary);
        border-bottom: 1px solid var(--color-text-secondary);
        box-sizing: border-box;
        @include apply-at-md-size {
          border-radius: 0;
          border: none;
          border-right: 1px solid var(--color-text-secondary);
        }

        &__title {
          @include flex-items-center(0.5rem);
          font-size: $font-size-regular;
          font-weight: 500;
          svg {
            @include box(1rem);
          }
        }
        &__option {
          @include flex-center;
        }
        .activeMale {
          color: #38d3e1;
          border-color: #38d3e1;
          background-color: transparent;
          .sex__label {
            font-weight: 700;
          }
        }
        .activeFemale {
          color: #ee18ae;
          border-color: #ee18ae;
          background-color: transparent;
          .sex__label {
            font-weight: 700;
          }
        }
        &__item {
          cursor: pointer;
          @include flex-center-column;
          border: 1px solid var(--color-text-primary);
          box-sizing: border-box;
          padding: 0.25rem;
          svg {
            @include box(3rem);
          }
        }
        &__male {
          border-bottom-left-radius: 1rem;
          border-top-left-radius: 1rem;
          &:hover {
            background-color: #38d3e1;
            border-color: #38d3e1;
            color: var(--color-bg-primary);
          }
        }
        &__female {
          border-top-right-radius: 1rem;
          border-bottom-right-radius: 1rem;
          &:hover {
            background-color: #ee18ae;
            border-color: #ee18ae;
            color: var(--color-bg-primary);
          }
        }
        &__label {
          font-size: $font-size-tiny;
        }
      }
      .dob {
        @include flex-center-column(0.5rem);
        width: 100%;
        padding: 0.75rem;
        border-bottom: 1px solid var(--color-text-secondary);
        box-sizing: border-box;
        @include apply-at-md-size {
          border: none;
        }
        &__label {
          @include flex-items-center(0.5rem);
          font-size: $font-size-regular;
          font-weight: 500;
          svg {
            @include box(1rem);
          }
        }
        &__input {
          @include input;
        }
        .wrapper {
          position: relative;
          .calendar {
            @include box(1rem);
            position: absolute;
            right: 8px;
            top: 50%;
            transform: translateY(-50%);
            z-index: 0;
          }
        }
      }
    }
    .measurements {
      @include flex-center-column(0.5rem);
      width: 100%;
      padding: 0.75rem;
      border-bottom: 1px solid var(--color-text-secondary);
      box-sizing: border-box;
      &__title {
        @include flex-items-center(0.25rem);
        font-weight: 500;
        svg {
          @include box(1.5rem);
        }
      }
      .measurement {
        @include flex-center(3rem);
        width: 100%;
        &__item {
          @include flex-center-column(0.5rem);
          .label {
            @include flex-items-center(0.5rem);
            font-size: $font-size-regular;
            font-weight: 500;
            svg {
              @include box(1rem);
            }
          }
          .input {
            @include input;
            width: 6rem;
            padding-right: 1.75rem;
            box-sizing: border-box;
          }
          .wrapper {
            position: relative;
            @include flex-center;
            .adornment {
              @include flex-center;
              font-size: $font-size-small;
              height: 100%;
              z-index: 2;
              position: absolute;
              right: 5%;
              top: 50%;
              transform: translateY(-50%);
            }
          }
        }
      }
    }
  }
}

input[type='date']::-webkit-calendar-picker-indicator {
  opacity: 0;
  z-index: 100;
  scale: 1.5;
  cursor: pointer;
}

input[type='number']::-webkit-inner-spin-button,
input[type='number']::-webkit-outer-spin-button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  margin: 0;
}

input[type='number'] {
  appearance: none;
  -moz-appearance: textfield;
}
