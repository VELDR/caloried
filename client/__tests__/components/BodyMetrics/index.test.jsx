import BodyMetrics from '@components/forms/BodyMetrics';
import { render, fireEvent } from '@utils/testHelper';

let wrapper;

const mockMetrics = {
  sex: 'male',
  dob: '1990-01-01',
  height: 180,
  weight: 75,
};

const mockProps = {
  metrics: mockMetrics,
  intl: { formatMessage: jest.fn() },
};

beforeEach(() => {
  wrapper = render(<BodyMetrics {...mockProps} />);
});

describe('BodyMetrics Component', () => {
  it('should render the component correctly', () => {
    const { getByTestId } = wrapper;

    expect(getByTestId('body-metrics')).toBeInTheDocument();
  });

  it('should update metrics data on user interaction', () => {
    const { getByTestId } = wrapper;

    const dobInput = getByTestId('metrics-dob');
    const heightInput = getByTestId('metrics-height');
    const weightInput = getByTestId('metrics-weight');
    fireEvent.click(getByTestId('sex-option-female'));
    fireEvent.change(dobInput, { target: { value: '1995-05-05' } });
    fireEvent.change(heightInput, { target: { value: 170 } });
    fireEvent.change(weightInput, { target: { value: 65 } });

    expect(getByTestId('sex-option-female')).toHaveClass('activeFemale');
    expect(dobInput.value).toBe('1995-05-05');
    expect(heightInput.value).toBe('170');
    expect(weightInput.value).toBe('65');
  });

  it('should match with snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
