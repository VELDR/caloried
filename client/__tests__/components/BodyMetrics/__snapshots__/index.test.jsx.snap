// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyMetrics Component should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form"
        data-testid="body-metrics"
      >
        <div
          class="header"
          data-testid="form-header"
        >
          <div
            class="header__title"
          >
            Fisik Tubuh
          </div>
          <div
            class="header__description"
          >
            Kami membutuhkan data fisik anda.
          </div>
        </div>
        <div
          class="metrics"
        >
          <div
            class="attributes"
          >
            <div
              class="sex"
            >
              <div
                class="sex__title"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="WcIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5.5 22v-7.5H4V9c0-1.1.9-2 2-2h3c1.1 0 2 .9 2 2v5.5H9.5V22h-4zM18 22v-6h3l-2.54-7.63C18.18 7.55 17.42 7 16.56 7h-.12c-.86 0-1.63.55-1.9 1.37L12 16h3v6h3zM7.5 6c1.11 0 2-.89 2-2s-.89-2-2-2-2 .89-2 2 .89 2 2 2zm9 0c1.11 0 2-.89 2-2s-.89-2-2-2-2 .89-2 2 .89 2 2 2z"
                  />
                </svg>
                Jenis Kelamin
              </div>
              <div
                class="sex__option"
              >
                <div
                  class="sex__item sex__male "
                  data-testid="sex-option-male"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="MaleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M9.5 11c1.93 0 3.5 1.57 3.5 3.5S11.43 18 9.5 18 6 16.43 6 14.5 7.57 11 9.5 11zm0-2C6.46 9 4 11.46 4 14.5S6.46 20 9.5 20s5.5-2.46 5.5-5.5c0-1.16-.36-2.23-.97-3.12L18 7.42V10h2V4h-6v2h2.58l-3.97 3.97C11.73 9.36 10.66 9 9.5 9z"
                    />
                  </svg>
                  <div
                    class="sex__label"
                  >
                    Laki-laki
                  </div>
                </div>
                <div
                  class="sex__item sex__female "
                  data-testid="sex-option-female"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="FemaleIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M17.5 9.5C17.5 6.46 15.04 4 12 4S6.5 6.46 6.5 9.5c0 2.7 1.94 4.93 4.5 5.4V17H9v2h2v2h2v-2h2v-2h-2v-2.1c2.56-.47 4.5-2.7 4.5-5.4zm-9 0C8.5 7.57 10.07 6 12 6s3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5z"
                    />
                  </svg>
                  <div
                    class="sex__label"
                  >
                    Perempuan
                  </div>
                </div>
              </div>
            </div>
            <div
              class="dob"
            >
              <div
                class="dob__label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CakeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 6c1.11 0 2-.9 2-2 0-.38-.1-.73-.29-1.03L12 0l-1.71 2.97c-.19.3-.29.65-.29 1.03 0 1.1.9 2 2 2zm4.6 9.99-1.07-1.07-1.08 1.07c-1.3 1.3-3.58 1.31-4.89 0l-1.07-1.07-1.09 1.07C6.75 16.64 5.88 17 4.96 17c-.73 0-1.4-.23-1.96-.61V21c0 .55.45 1 1 1h16c.55 0 1-.45 1-1v-4.61c-.56.38-1.23.61-1.96.61-.92 0-1.79-.36-2.44-1.01zM18 9h-5V7h-2v2H6c-1.66 0-3 1.34-3 3v1.54c0 1.08.88 1.96 1.96 1.96.52 0 1.02-.2 1.38-.57l2.14-2.13 2.13 2.13c.74.74 2.03.74 2.77 0l2.14-2.13 2.13 2.13c.37.37.86.57 1.38.57 1.08 0 1.96-.88 1.96-1.96V12C21 10.34 19.66 9 18 9z"
                  />
                </svg>
                 
                Tanggal Lahir
              </div>
              <div
                class="wrapper"
              >
                <input
                  class="dob__input"
                  data-testid="metrics-dob"
                  name="dob"
                  type="date"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium calendar css-i4bv87-MuiSvgIcon-root"
                  data-testid="CalendarTodayIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="measurements"
          >
            <div
              class="measurements__title"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SquareFootIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m17.66 17.66-1.06 1.06-.71-.71 1.06-1.06-1.94-1.94-1.06 1.06-.71-.71 1.06-1.06-1.94-1.94-1.06 1.06-.71-.71 1.06-1.06L9.7 9.7l-1.06 1.06-.71-.71 1.06-1.06-1.94-1.94-1.06 1.06-.71-.71 1.06-1.06L4 4v14c0 1.1.9 2 2 2h14l-2.34-2.34zM7 17v-5.76L12.76 17H7z"
                />
              </svg>
               
              Pengukuran
            </div>
            <div
              class="measurement"
            >
              <div
                class="measurement__item"
              >
                <div
                  class="label"
                >
                  Tinggi Badan
                </div>
                <div
                  class="wrapper"
                >
                  <input
                    class="input"
                    data-testid="metrics-height"
                    name="height"
                    type="number"
                    value=""
                  />
                  <div
                    class="adornment"
                  >
                    cm
                  </div>
                </div>
              </div>
              <div
                class="measurement__item"
              >
                <div
                  class="label"
                >
                  Berat Badan
                </div>
                <div
                  class="wrapper"
                >
                  <input
                    class="input"
                    data-testid="metrics-weight"
                    name="weight"
                    type="number"
                    value=""
                  />
                  <div
                    class="adornment"
                  >
                    kg
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="footer"
          data-testid="form-footer"
        >
          <button
            class="button footer__right footer__button"
            data-testid="primary-button"
            type="submit"
          >
            Mulai
             
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="EastIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m15 5-1.41 1.41L18.17 11H2v2h16.17l-4.59 4.59L15 19l7-7-7-7z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form"
      data-testid="body-metrics"
    >
      <div
        class="header"
        data-testid="form-header"
      >
        <div
          class="header__title"
        >
          Fisik Tubuh
        </div>
        <div
          class="header__description"
        >
          Kami membutuhkan data fisik anda.
        </div>
      </div>
      <div
        class="metrics"
      >
        <div
          class="attributes"
        >
          <div
            class="sex"
          >
            <div
              class="sex__title"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="WcIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5.5 22v-7.5H4V9c0-1.1.9-2 2-2h3c1.1 0 2 .9 2 2v5.5H9.5V22h-4zM18 22v-6h3l-2.54-7.63C18.18 7.55 17.42 7 16.56 7h-.12c-.86 0-1.63.55-1.9 1.37L12 16h3v6h3zM7.5 6c1.11 0 2-.89 2-2s-.89-2-2-2-2 .89-2 2 .89 2 2 2zm9 0c1.11 0 2-.89 2-2s-.89-2-2-2-2 .89-2 2 .89 2 2 2z"
                />
              </svg>
              Jenis Kelamin
            </div>
            <div
              class="sex__option"
            >
              <div
                class="sex__item sex__male "
                data-testid="sex-option-male"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MaleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M9.5 11c1.93 0 3.5 1.57 3.5 3.5S11.43 18 9.5 18 6 16.43 6 14.5 7.57 11 9.5 11zm0-2C6.46 9 4 11.46 4 14.5S6.46 20 9.5 20s5.5-2.46 5.5-5.5c0-1.16-.36-2.23-.97-3.12L18 7.42V10h2V4h-6v2h2.58l-3.97 3.97C11.73 9.36 10.66 9 9.5 9z"
                  />
                </svg>
                <div
                  class="sex__label"
                >
                  Laki-laki
                </div>
              </div>
              <div
                class="sex__item sex__female "
                data-testid="sex-option-female"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="FemaleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17.5 9.5C17.5 6.46 15.04 4 12 4S6.5 6.46 6.5 9.5c0 2.7 1.94 4.93 4.5 5.4V17H9v2h2v2h2v-2h2v-2h-2v-2.1c2.56-.47 4.5-2.7 4.5-5.4zm-9 0C8.5 7.57 10.07 6 12 6s3.5 1.57 3.5 3.5S13.93 13 12 13s-3.5-1.57-3.5-3.5z"
                  />
                </svg>
                <div
                  class="sex__label"
                >
                  Perempuan
                </div>
              </div>
            </div>
          </div>
          <div
            class="dob"
          >
            <div
              class="dob__label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CakeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 6c1.11 0 2-.9 2-2 0-.38-.1-.73-.29-1.03L12 0l-1.71 2.97c-.19.3-.29.65-.29 1.03 0 1.1.9 2 2 2zm4.6 9.99-1.07-1.07-1.08 1.07c-1.3 1.3-3.58 1.31-4.89 0l-1.07-1.07-1.09 1.07C6.75 16.64 5.88 17 4.96 17c-.73 0-1.4-.23-1.96-.61V21c0 .55.45 1 1 1h16c.55 0 1-.45 1-1v-4.61c-.56.38-1.23.61-1.96.61-.92 0-1.79-.36-2.44-1.01zM18 9h-5V7h-2v2H6c-1.66 0-3 1.34-3 3v1.54c0 1.08.88 1.96 1.96 1.96.52 0 1.02-.2 1.38-.57l2.14-2.13 2.13 2.13c.74.74 2.03.74 2.77 0l2.14-2.13 2.13 2.13c.37.37.86.57 1.38.57 1.08 0 1.96-.88 1.96-1.96V12C21 10.34 19.66 9 18 9z"
                />
              </svg>
               
              Tanggal Lahir
            </div>
            <div
              class="wrapper"
            >
              <input
                class="dob__input"
                data-testid="metrics-dob"
                name="dob"
                type="date"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium calendar css-i4bv87-MuiSvgIcon-root"
                data-testid="CalendarTodayIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="measurements"
        >
          <div
            class="measurements__title"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SquareFootIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m17.66 17.66-1.06 1.06-.71-.71 1.06-1.06-1.94-1.94-1.06 1.06-.71-.71 1.06-1.06-1.94-1.94-1.06 1.06-.71-.71 1.06-1.06L9.7 9.7l-1.06 1.06-.71-.71 1.06-1.06-1.94-1.94-1.06 1.06-.71-.71 1.06-1.06L4 4v14c0 1.1.9 2 2 2h14l-2.34-2.34zM7 17v-5.76L12.76 17H7z"
              />
            </svg>
             
            Pengukuran
          </div>
          <div
            class="measurement"
          >
            <div
              class="measurement__item"
            >
              <div
                class="label"
              >
                Tinggi Badan
              </div>
              <div
                class="wrapper"
              >
                <input
                  class="input"
                  data-testid="metrics-height"
                  name="height"
                  type="number"
                  value=""
                />
                <div
                  class="adornment"
                >
                  cm
                </div>
              </div>
            </div>
            <div
              class="measurement__item"
            >
              <div
                class="label"
              >
                Berat Badan
              </div>
              <div
                class="wrapper"
              >
                <input
                  class="input"
                  data-testid="metrics-weight"
                  name="weight"
                  type="number"
                  value=""
                />
                <div
                  class="adornment"
                >
                  kg
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="footer"
        data-testid="form-footer"
      >
        <button
          class="button footer__right footer__button"
          data-testid="primary-button"
          type="submit"
        >
          Mulai
           
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EastIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m15 5-1.41 1.41L18.17 11H2v2h16.17l-4.59 4.59L15 19l7-7-7-7z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
